src/main/java/uk/gov/ons/census/action/messaging/FulfilmentRequestReceiver.java

```diff
@@ -0,0 +1,169 @@
+package uk.gov.ons.census.action.messaging;
+
+import com.godaddy.logging.Logger;
+import com.godaddy.logging.LoggerFactory;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.integration.annotation.MessageEndpoint;
+import org.springframework.integration.annotation.ServiceActivator;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.ons.census.action.client.CaseClient;
+import uk.gov.ons.census.action.model.dto.PrintFileDto;
+import uk.gov.ons.census.action.model.dto.ResponseManagementEvent;
+import uk.gov.ons.census.action.model.dto.UacQidDTO;
+import uk.gov.ons.census.action.model.entity.ActionType;
+import uk.gov.ons.census.action.model.entity.Case;
+import uk.gov.ons.census.action.model.repository.CaseRepository;
+
+@MessageEndpoint
+public class FulfilmentRequestReceiver {
+  private static final Logger log = LoggerFactory.getLogger(FulfilmentRequestReceiver.class);
+  private final RabbitTemplate rabbitTemplate;
+  private final CaseClient caseClient;
+  private final CaseRepository caseRepository;
+
+  @Value("${queueconfig.outbound-exchange}")
+  private String outboundExchange;
+
+  @Value("${queueconfig.outbound-printer-routing-key}")
+  private String outboundPrinterRoutingKey;
+
+  public FulfilmentRequestReceiver(
+      RabbitTemplate rabbitTemplate, CaseClient caseClient, CaseRepository caseRepository) {
+    this.rabbitTemplate = rabbitTemplate;
+    this.caseClient = caseClient;
+    this.caseRepository = caseRepository;
+  }
+
+  @Transactional
+  @ServiceActivator(inputChannel = "actionFulfilmentInputChannel")
+  public void receiveEvent(ResponseManagementEvent event) {
+    Case fulfilmentCase = fetchFulfilmentCase(event);
+    String fulfilmentCode = event.getPayload().getFulfilmentRequest().getFulfilmentCode();
+
+    Optional<ActionType> actionType;
+    try {
+      actionType = determineActionType(fulfilmentCode);
+    } catch (UnknownFulfilmentException e) {
+      log.with("fulfilmentCode", fulfilmentCode).warn("Unexpected fulfilment code received");
+      return;
+    }
+    if (actionType.isEmpty()) {
```

Comment:
whaaat? Looks like we're handing things in a million different ways? Why do the stuff above if we're just going to do this?